Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Paged/WalletSolanaProviderScript.js

```diff
@@ -60,27 +60,79 @@ window.__firefox__.execute(function($, $Object, $Function, $Array) {
           )
       }))
     })
-    /* <solanaWeb3.Transaction> ->
-      {transaction: <solanaWeb3.Transaction>,
-       serializedMessage: <base58 encoded string>,
-       signatures: [{publicKey: <base58 encoded string>, signature: <Buffer>}]} */
+    /*
+     solanaWeb3.Transaction | solanaWeb3.VersionedTransaction
+     ->
+     [UInt8]
+     */
+    let serializedMessageFromTx = $(function(transaction) {
+      // VersionedTransaction (v0) : Transaction (legacy)
+      let serializeMessageBuffer = transaction.message ? transaction.message.serialize() : transaction.serializeMessage();
+      return [...serializeMessageBuffer]; // Buffer to Array
+    })
+    /*
+     solanaWeb3.Transaction | solanaWeb3.VersionedTransaction
+     ->
+     [{publicKey: <base58 encoded string>, signature: [UInt8]}]
+     */
+    let signaturesPubKeyPairsFromTx = $(function(transaction) {
+      if (transaction.message) { // VersionedTransaction (v0)
+        const signatures = transaction.signatures;
+        const versionedMessage = transaction.message;
+        const staticAccountKeys = versionedMessage.staticAccountKeys;
+
+        const signaturePubkeyObjects = [];
+        for (let i = 0; i < signatures.length; i++) {
+          const signature = signatures[i];
+          const publicKey = staticAccountKeys[i];
+
+          const obj = $Object.create(null, undefined);
+          obj.publicKey = publicKey.toBase58();
+          obj.signature = signature;
+          signaturePubkeyObjects[i] = $.extensiveFreeze(obj, freezeExceptions);
+        }
+
+        return signaturePubkeyObjects;
+      } else { // Transaction (legacy)
+        let convertSignaturePubkeyTuple = $(function(signaturePubkeyTuple) {
+          const obj = $Object.create(null, undefined);
+          obj.publicKey = signaturePubkeyTuple.publicKey.toBase58();
+          const signatureBuffer = signaturePubkeyTuple.signature
+          if (signatureBuffer) {
+            obj.signature = [...signatureBuffer]; // Buffer to Array
+          } else {
+            obj.signature = [];
+          }
+          return $.extensiveFreeze(obj, freezeExceptions);
+        })
+        return transaction.signatures.map(convertSignaturePubkeyTuple);
```

Comment:
This `map` function can be overridden by page. Can we use original `$Array` object here instead? cc @Brandon-T

Reply:
Thanks! I've updated this in [efb63bf](https://github.com/brave/brave-ios/pull/7369/commits/efb63bf3053b28fa265dda25d7eb7842a876284b), and it two spots in `signAllTransactions ` that were also using `map` function.